cli/src/action/registry/mod.rs

```diff
@@ -177,3 +182,161 @@ impl Action for RegistryGenerateAction {
+                .map(|kv| {
+                    let mut kv_iter = kv.splitn(2, '=');
+
+                    let key = kv_iter
+                        .next()
+                        .expect("str::split cannot return an empty iterator")
+                        .to_string();
+                    if key.is_empty() {
+                        return Err(CliError::ActionError(
+                            "Empty '--metadata' argument detected".into(),
+                        ));
+                    }
+
+                    let value = kv_iter
+                        .next()
+                        .ok_or_else(|| {
+                            CliError::ActionError(format!(
+                                "Missing value for metadata key '{}'",
+                                key
+                            ))
+                        })?
+                        .to_string();
+                    if value.is_empty() {
+                        return Err(CliError::ActionError(format!(
+                            "Empty value detected for metadata key '{}'",
+                            key
+                        )));
+                    }
+
+                    Ok((key, value))
+                })
+                .collect::<Result<_, _>>()?
+        } else {
+            Default::default()
+        };
+
+        if args.is_present("from_remote") {
+            let remote_node = client.get_node(&identity)?.ok_or_else(|| {
+                CliError::ActionError("Unable to retrieve node from remote".into())
+            })?;
+            let endpoints = match args.values_of("endpoint") {
+                Some(endpoints) => endpoints.map(String::from).collect::<Vec<String>>(),
+                None => remote_node.endpoints,
+            };
+            let keys = match args.values_of("key") {
+                Some(keys) => keys
+                    .map(|key_file| read_private_key(key_file))
+                    .collect::<Result<_, _>>()?,
+                None => remote_node.keys,
+            };
+
+            let mut node_metadata: HashMap<String, String> = HashMap::new();
+            if args.is_present("metadata") {
+                for (key, value) in metadata.iter() {
+                    node_metadata.insert(key.to_string(), value.to_string());
+                }
+            } else {
+                node_metadata = remote_node.metadata;
+            }
+
+            let mut node_builder = Node::builder(identity)
+                .with_endpoints(endpoints)
+                .with_display_name(display_name)
+                .with_keys(keys);
+
+            for (key, value) in node_metadata {
+                node_builder = node_builder.with_metadata(key, value);
+            }
+
+            let node = node_builder
+                .build()
+                .map_err(|err| CliError::ActionError(format!("Unable to build node: {}", err)))?;
+
+            if !args.is_present("dry_run") {
+                client.update_node(&node)?;
```

Comment:
I think we discussed how this should use `add_node` but can't because of the way the registry and its REST API are currently designed; the changes we discussed to correct this should be put in the feature doc for the registry CLI.

Reply:
Added the commit with these changes